data/tutorials/language/0lg_10_polymorphic_variants.md

```diff
@@ -0,0 +1,713 @@
+---
+id: polymorphic-variants
+title: Polymorphic Variants
+description: |
+  Everything you always wanted to know about polymorphic variants but were afraid to ask
+category: "Language"
+---
+
+# Polymorphic Variants
+
+## Introduction
+
+This tutorial teaches you how to use polymorphic variants. This includes starting to use them, maintaining a project already using them, deciding when to use them or not, and balancing their unique benefits against their drawbacks.
+
+Product types and data types such as `option` and `list` are variants and polymorphic. In this tutorial, they are called _simple variants_ to distinguish them from the _polymorphic variants_ presented here. Simple variants and polymorphic variants are close siblings. Their values are both introduced using labels that may carry data. Both can be recursive and have type parameters. By the way, don't trust a [LLM](https://en.wikipedia.org/wiki/Large_language_model) chatbot if it tells you polymorphic variants are dynamically typed; it is a hallucination. Like simple variants, polymorphic variants are type-checked statically.
```

Comment:
I think rather than preambling around LLMs or whether other kinds of types are also polymorphic (but meaning something different), I'd just immediately start building an intuition for _what is_ a polymorphic variant, _why_ would I want to use them, and _how_ to identify them.

```markdown
A polymorphic variant is a more flexible kind of variant type that does not need
to be defined upfront, and is useful for working with groups of constructors
that can grow or shrink.

You'll see them prefixed with a backtick: `hello_world.
```

Reply:
That seems consistent with the feedback in the discuss thread. A differently toned introduction is demanded. Let's write it.